### ISSUE TITLE
getReddit throws exception

### ISSUE THREAD

**Issue by vimuel** [sellerlabs/haskell-workshop#8]
```
*FinalReddit> getReddit "haskell"
*** Exception: not enough input
```
Any ideas?

**Comment by parsonsmatt** [comment]
Here's the definition of `getReddit`:

```haskell
getReddit :: String -> IO Listing
getReddit subreddit = do
  let url = "http://www.reddit.com/r/" <> subreddit <> "/hot/.json?count=25"
  response <- simpleHTTP (getRequest url)
  body <- getResponseBody response
  case eitherDecode (cs body) of
    Right listing -> pure listing
    Left e -> error e
```

My guess is that `eitherDecode` is failing when trying to parse the input.

**Comment by vimuel** [comment]
Exactly. But why?

Is it working on your end?

**Comment by parsonsmatt** [comment]
I haven't looked at the response in quite some time. I'd guess that Reddit changed their API or the URL structure such that the parsing methods don't work anymore.

**Comment by vimuel** [comment]
OK, so it looks like Reddit can only be accessed via https now, which is not supported by simpleHTTP. I've tried using http-conduit which supports https, but I haven't managed yet.

**Comment by 5outh** [comment]
Oh, that is a bummer!

There are a few HTTP clients for haskell that support https without too much overhead. Off the top of my head, there's [wreq, which has a nice tutorial](http://www.serpentine.com/wreq/tutorial.html) and shouldn't be too tough to drop-in replace.

**Comment by vimuel** [comment]
Wohoo! Thanks for that info — I got it to work in a breeze.

Would it be helpful if I opened a pull-request with updated instructions in the README?

**Comment by 5outh** [comment]
@vimuel Sure! I don't want to make any edits to the actual code base in order to maintain consistency with the video, but a disclaimer in the README sounds like a good compromise.

**Comment by vimuel** [comment]
Yeah, I will open a pull-request for an updated README if I manage to get everything running.

PS: I figured it was better to leave this issue open so that others that try to follow the tutorial easily find the solution.

**Comment by Lightyagami1** [comment]
Hi @5outh , could you please mention which version of lens and wreq are needed ?

**Comment by Lightyagami1** [comment]
I added these in extra dependency,
- lens-4.13
- wreq-0.4.1.0
- containers-0.5.6.2
but on executing `stack build`, I get these errors
Could not find module ‘Network.Wreq’
Could not find module ‘Control.Lens.Getter’
Could you tell why they are not being build ?

**Comment by 5outh** [comment]
@Lightyagami1 A `stack build` with no arguments from the root directory works for me - can you paste the full output? You shouldn't need any extra dependencies.

I should mention that I don't know how well this project has aged. It's possible that the reddit API has drifted and the project no longer works at all :sweat_smile:
